nautobot/core/celery/log.py

```diff
@@ -0,0 +1,30 @@
+import logging
+
+from celery import current_task
+
+
+class NautobotDatabaseHandler(logging.Handler):
+    """Custom logging handler to log messages to JobLogEntry database entries."""
+
+    def emit(self, record):
+        if current_task is None:
+            return
+
+        from nautobot.extras.models.jobs import JobResult
+
+        try:
+            self.format(record)
+
+            if record.task_id == "???":
+                return
+            job_result = JobResult.objects.filter(id=record.task_id)
+            if not job_result.exists():
+                return
```

Comment:
It would probably have to be an exception or a write to stdout/stderr since trying to log here might go into an infinite loop. It's probably safe to raise an exception here since I can't think of a way to get to this line of code without the job result being created.